Sources/Target/Common/ThreadBase.cpp

```diff
@@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
   action(state);
   return writeCPUState(state);
 }
+
+ErrorCode ThreadBase::beforeResume() {
+  if (!_process->isAlive())
+    return kErrorProcessNotFound;
+
+  BreakpointManager *bpm = _process->hardwareBreakpointManager();
+  if (bpm != nullptr) {
+    bpm->enable();
```

Comment:
What LLDB does doesn't really matter here. You could just get a
HardwareBreakpointManager pointer instead of a generic BreakpointManager
pointer and use that given that an API exists there, no?

On Tuesday, August 9, 2016, Francis Ricci <notifications@github.com> wrote:

> In Sources/Target/Common/ThreadBase.cpp
> <https://github.com/facebook/ds2/pull/243#discussion_r74099416>:
>
> > @@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
> >    action(state);
> >    return writeCPUState(state);
> >  }
> > +
> > +ErrorCode ThreadBase::beforeResume() {
> > +  if (!_process->isAlive())
> > +    return kErrorProcessNotFound;
> > +
> > +  BreakpointManager *bpm = _process->hardwareBreakpointManager();
> > +  if (bpm != nullptr) {
> > +    bpm->enable();
>
> A function to enable breakpoints on a specific thread currently doesn't
> exist in the BreakpointManager API, since software breakpoints don't care
> about threads. I could add it, but given that thread-specific hardware
> breakpoints aren't a thing in lldb, I don't think it's necessary, and just
> complicates the API.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/2a73b10db4691d10fd51d07ec46fbded7b36f6af..b32f1128fa8ceb0010cf5f79262d062aca477162#r74099416>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADHbNPGgbZm4wGt4jnVepP1XM_W375Gks5qeLIPgaJpZM4H9FRy>
> .
>


-- 
Stephane Sezer